ConsoleApp2/ConsoleApp2/SOLID/SingleResponsibilityPrinciple.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace LearnBasics.Sandbox.SOLID
+{
+    class SingleResponsibilityPrinciple
+    {
+        public void Example()
+        {
+            var store = new MobileStore(
+                new ConsolePhoneReader(), new GeneralPhoneBinder(),
+                new GeneralPhoneValidator(), new TextPhoneSaver());
+            store.Process();
+        }
+    }
+    class Phone
+    {
+        public string Model { get; set; }
+        public int Price { get; set; }
+    }
+
+    class MobileStore
+    {
+        List<Phone> phones = new List<Phone>();
+
+        public IPhoneReader Reader { get; set; }
+        public IPhoneBinder Binder { get; set; }
+        public IPhoneValidator Validator { get; set; }
+        public IPhoneSaver Saver { get; set; }
+
+        public MobileStore(IPhoneReader reader, IPhoneBinder binder, IPhoneValidator validator, IPhoneSaver saver)
+        {
+            this.Reader = reader;
+            this.Binder = binder;
+            this.Validator = validator;
+            this.Saver = saver;
+        }
+
+        public void Process()
+        {
+            string[] data = Reader.GetInputData();
+            var phone = Binder.CreatePhone(data);
+            if (Validator.IsValid(phone))
+            {
+                phones.Add(phone);
+                Saver.Save(phone, "store.txt");
+                Console.WriteLine("Данные успешно обработаны");
+            }
+            else
+            {
+                Console.WriteLine("Некорректные данные");
+            }
+        }
+    }
+
+    interface IPhoneReader
+    {
+        string[] GetInputData();
+    }
+    class ConsolePhoneReader : IPhoneReader
+    {
+        public string[] GetInputData()
+        {
+            Console.WriteLine("Введите модель:");
+            string model = Console.ReadLine();
+            Console.WriteLine("Введите цену:");
+            string price = Console.ReadLine();
+            return new string[] { model, price };
```

Comment:
better to use Model for model & price fields

Reply:
done